crates/proof-impl/checkpoint/src/lib.rs

```diff
@@ -21,7 +21,7 @@ pub struct ChainStateSnapshot {
 
 #[derive(Debug, Clone, BorshSerialize, BorshDeserialize)]
 pub struct L2BatchProofOutput {
-    pub deposits: Vec<DepositInfo>,
+    pub deposits: Vec<ELDepositData>,
```

Comment:
Can you elaborate on what this field represents now?  Having the indexes in here is redundant since they'll always be ordered, etc..  This type wasn't meant to exist as a long term solution, it was part of the `Op` stopgap.  What goal are we trying to achieve here?

Since we can figure out the indexes from other externally observed data, we can construct everything from purely the index that was processed by looking at the deposit intents queue on the exec env.

Reply:
The idea is to verify that the deposits identified from the L1 scan match those made on the EL layer. This ensures that the sequencer did not overlook any deposit transactions settled on L1.

On the current CL-EL implementation, we are passing the deposits as Eth-POS `Withdrawals` while building the block. So, from the EL-CL proof we can recover `ELDepositData` not `DepositInfo` so following changes were done.

Reply:
Ah we gotta rework this post devnet.